From 74845d6cd3abcf7586f32e222131849c66cc6ad8 Mon Sep 17 00:00:00 2001
From: Emmanuele Bassi <ebassi@gnome.org>
Date: Thu, 1 Apr 2021 22:58:34 +0100
Subject: [PATCH] Fix the GCC check in graphene-config.h
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

We want GCC ≥ 4.9 on non-ARM architectures, so we need to check for:

 - GCC
 - GCC ≥ 5 or GCC == 4.9
 - !ARM
---
 include/graphene-config.h.meson | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/graphene-config.h.meson b/include/graphene-config.h.meson
index ab72d53..949eee7 100644
--- a/include/graphene-config.h.meson
+++ b/include/graphene-config.h.meson
@@ -23,7 +23,7 @@ extern "C" {
 #mesondefine GRAPHENE_HAS_ARM_NEON
 #  endif
 
-#  if defined(__GNUC__) && (__GNUC__ >= 4 && __GNUC_MINOR__ >= 9) && !defined(__arm__)
+#  if defined(__GNUC__) && (__GNUC__ >= 5 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 9)) && !defined(__arm__)
 #mesondefine GRAPHENE_HAS_GCC
 #  endif
 
